src/i_oalmusic.c

```diff
@@ -304,20 +336,18 @@ static void I_OAL_SetMusicVolume(int volume)
         return;
     }
 
-    ALfloat gain = (ALfloat)volume / 15.0f;
+    player.gain = (ALfloat)volume / 15.0f;
 
     if (active_module == &stream_opl_module)
     {
-        gain *= (ALfloat)DB_TO_GAIN(opl_gain);
+        player.gain *= (ALfloat)DB_TO_GAIN(opl_gain);
     }
 #if defined(HAVE_FLUIDSYNTH)
     else if (active_module == &stream_fl_module)
     {
-        gain *= (ALfloat)DB_TO_GAIN(fl_gain);
+        player.gain *= (ALfloat)DB_TO_GAIN(fl_gain);
     }
 #endif
-
-    alSourcef(player.source, AL_GAIN, gain);
```

Comment:
I think it won't hurt to call this once to initialize the `player.source` gain level outside of the ebur128 code path.

Reply:
Then the beginning of the track will be loud. I think we should adjust the gain only after normalization.

Reply:
Okay, so we are sending the music with a 400 ms delay to have time to adjust loudness?

Reply:
No, we render music ahead of time for the OpenAL stream.

Normalization works for all music players (including Fluidsynth and OPL) except native MIDI, because it's real time only.

Reply:
Will this mean that all quiet parts of a song will become louder and all loud parts will get quieter, so the song will be about the same volume all along?

Reply:
> Will this mean that all quiet parts of a song will become louder and all loud parts will get quieter, so the song will be about the same volume all along?

It should be smart about it. Seems to work correctly, but more testing is needed.

Reply:
> It should be smart about it. Seems to work correctly, but more testing is needed.

In any case, I think we should make this optional, and not even config only. This feature has the potential to break some carefully crafted tension build-up by MUSINFO object placement with music tracks which feature sensible volume changes.